infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -148,54 +154,8 @@ public LayoutServer(ServerContext serverContext) {
         // QuickCheck: Create the distributed Erlang message handling threads
         Object test_mode = opts.get("--quickcheck-test-mode");
         if (test_mode != null && (Boolean) test_mode) {
-            QuickCheckMode qm = new QuickCheckMode(opts);
-        }
-    }
-
-    //TODO need to figure out if we need to send the complete Rank object in the responses
-    @Override
-    public synchronized void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        if (isShutdown()) return;
-        if (r.getClass() == NettyServerRouter.class) {
-            NettyServerRouter nsr = (NettyServerRouter) r;
-            if (!nsr.validateEpoch(msg, ctx)) {
-                // RACE!  The epoch changed in between the epoch validation that
-                //        NettyServerRouter performed and the current instant.
-                //        We're now in synchronized object territory, so the epoch
-                //        cannot change beyond this point, but for this case, it's
-                //        too late.  The appropriate reply has been sent to the client.
-                return;
-            }
-        }
-        // This server has not been bootstrapped yet, ignore ALL requests except for LAYOUT_BOOTSTRAP
-        if (getCurrentLayout() == null && !msg.getMsgType().equals(CorfuMsgType.LAYOUT_BOOTSTRAP)) {
-            log.warn("Received message but not bootstrapped! Message={}", msg);
-            r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.LAYOUT_NOBOOTSTRAP));
-            return;
-        }
-        switch (msg.getMsgType()) {
-            case LAYOUT_REQUEST:
-                handleMessageLayoutRequest((CorfuPayloadMsg<Long>) msg, ctx, r);
-                break;
-            case LAYOUT_BOOTSTRAP:
-                handleMessageLayoutBootStrap((CorfuPayloadMsg<LayoutBootstrapRequest>)msg, ctx, r);
-                break;
-            case SET_EPOCH:
-                handleMessageSetEpoch((CorfuPayloadMsg<Long>) msg, ctx, r);
-                break;
-            case LAYOUT_PREPARE:
-                handleMessageLayoutPrepare((CorfuPayloadMsg<LayoutPrepareRequest>) msg, ctx, r);
-                break;
-            case LAYOUT_PROPOSE:
-                handleMessageLayoutPropose((CorfuPayloadMsg<LayoutProposeRequest>) msg, ctx, r);
-                break;
-            case LAYOUT_COMMITTED: {
-                handleMessageLayoutCommit((CorfuPayloadMsg<LayoutCommittedRequest>) msg, ctx, r);
-            }
-            break;
-            default:
-                log.warn("Unknown message type {} passed to handler!", msg.getMsgType());
-                throw new RuntimeException("Unsupported message passed to handler!");
+            // Disabled due to cyclic dependency...
+         //   QuickCheckMode qm = new QuickCheckMode(opts);
```

Comment:
It's because QuickCheckMode is a class which the runtime and the cmdlets depend on, so it needs to be in its own package, which the runtime and cmdlets would depend on. That doesn't seem like a good idea to me... Probably the right thing to do would be to remove QuickCheckMode from LayoutServer, and define the right interfaces, if we're going to continue using this class.